src/app/components/upload-card/UploadCardRenderer.tsx

```diff
@@ -53,6 +66,30 @@ export function UploadCardRenderer({
               <Text size="B300">Retry</Text>
             </Chip>
           )}
+          {file.type.startsWith('image') && (
+            <TooltipProvider
+              tooltip={
+                <Tooltip variant="SurfaceVariant">
+                  <Text>Spoiler Image</Text>
+                </Tooltip>
+              }
+              position="Top"
+              align="Center"
+            >
+              {(triggerRef) => (
+                <IconButton
+                  ref={triggerRef}
+                  onClick={toggleSpoiler}
+                  aria-label="Spoil Image"
+                  variant="SurfaceVariant"
+                  radii="Pill"
+                  size="300"
+                >
+                  <Icon src={metadata.markedAsSpoiler ? Icons.EyeBlind : Icons.Eye} size="200" />
```

Comment:
if found that changing the icon makes it a little confusing about the state of the spoiler. we should make it similar to how we have spoiler button in rich editor (Icons.EyeBlind with pressed state when acitve)